python/ray/air/execution/actor_manager.py

```diff
@@ -0,0 +1,244 @@
+
+        Events:
+            ``ActorStopped``: Emitted once the actor has been started. If the actor
+                has only been requested (but has not been started, yet),
+                this event won't be emitted.
+
+        Args:
+            actor_info: Actor info object of the actor to be removed.
+            resolve_futures: If True, will resolve associated futures (and emit
+                events) first before stopping the actor.
+            kill: If set, will forcefully terminate the actor instead of gracefully
+                scheduling termination.
+        """
+        raise NotImplementedError
+
+    def is_actor_started(self, actor_info: ActorInfo) -> bool:
+        """Returns True if the actor has been started.
+
+        Args:
+            actor_info: Actor info object of the actor to inquire state about.
+        """
+        raise NotImplementedError
+
+    def get_actor_resources(self, actor_info: ActorInfo) -> Optional[AcquiredResources]:
+        """Returns the acquired resources of an actor that has been started.
+
+        This will return ``None`` if the actor has not been started, yet.
+
+        Args:
+            actor_info: Actor info object of the actor to get resources for.
+        """
+        raise NotImplementedError
+
+    def schedule_task(
+        self,
+        actor_info: ActorInfo,
+        task_spec: Tuple[str, Iterable[Any], Dict[str, Any]],
+        result_cls: Type[FutureResult],
+    ) -> None:
+        """Track (asynchronous) future associated to a tracked actor.
+
+        This method will schedule the remote task and track its execution. Once the
+        future is resolved, it will be emitted as an event. The event
+        class will be ``result_cls``, which should extend ``FutureResult``.
+
+        The ``task_spec`` argument is submitted as a tuple of
+        ``(method_name, args, kwargs)``. This will result in the future
+        ``future = actor.method_name.remote(*args, **kwargs)`` to be scheduled on
+        the actor and tracked in the manager.
+
+        Events:
+            ``result_cls``: Emitted once the future resolved.
+            ``FutureFailed``: Emitted if the future resolution fails due to a
+                ``RayTaskError`, indicating method execution ran into an exception.
+            ``ActorFailed``: Emitted if the future resolution fails due to a
+                ``RayActorError``, indicating the actor died.
+
+        Args:
+            actor_info: Actor info object to track future for.
+            task_spec: Tuple of ``(method_name, args, kwargs)`` to be scheduled as
+                task futures on the actor.
+            result_cls: Class extending ``FutureResult`` to emit when the future
+                resolves successfully.
+
+        """
+        raise NotImplementedError
+
+    def schedule_sync_tasks(
+        self,
+        actors_tasks: Dict[ActorInfo, Tuple[str, Iterable[Any], Dict[str, Any]]],
+        result_cls: Type[FutureResult],
```

Comment:
any ways to specify timeout?

Reply:
We can add this. Is rllib using this already or is that future use?

Reply:
I'm holding back on adding this right now until we clarified the use